catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -123,18 +150,170 @@ public Fileset createFileset(
+            .withComment(comment)
+            .withFilesetType(type)
+            // Store the storageLocation to the store. If the "storageLocation" is null for
+            // managed fileset, Gravitino will get and store the location based on the
+            // catalog/schema's location.
+            .withStorageLocation(filesetPath.toString())
+            .withProperties(addManagedFlagToProperties(properties))
+            .withAuditInfo(
+                new AuditInfo.Builder()
+                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                    .withCreateTime(Instant.now())
+                    .build())
+            .build();
+
+    try {
+      store.put(filesetEntity, true /* overwrite */);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to create fileset " + ident, ioe);
+    }
+
+    return new HadoopFileset.Builder()
+        .withName(ident.name())
+        .withComment(comment)
+        .withType(type)
+        // The returned storageLocation is a calculated one. If the fileset is managed and
+        // storageLocation is null, then the calculated location is the
+        // schema location + fileset name.
+        .withStorageLocation(filesetPath.toString())
+        .withProperties(filesetEntity.properties())
+        .withAuditInfo(filesetEntity.auditInfo())
+        .build();
   }
 
   @Override
   public Fileset alterFileset(NameIdentifier ident, FilesetChange... changes)
       throws NoSuchFilesetException, IllegalArgumentException {
-    throw new UnsupportedOperationException();
+    try {
+      if (!store.exists(ident, Entity.EntityType.FILESET)) {
+        throw new NoSuchFilesetException("Fileset " + ident + " does not exist");
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load fileset " + ident, ioe);
+    }
+
+    try {
+      FilesetEntity updatedFilesetEntity =
+          store.update(
+              ident,
+              FilesetEntity.class,
+              Entity.EntityType.FILESET,
+              e -> updateFilesetEntity(ident, e, changes));
+
+      return new HadoopFileset.Builder()
+          .withName(updatedFilesetEntity.name())
+          .withComment(updatedFilesetEntity.comment())
+          .withType(updatedFilesetEntity.filesetType())
+          .withStorageLocation(updatedFilesetEntity.storageLocation())
+          .withProperties(updatedFilesetEntity.properties())
+          .withAuditInfo(updatedFilesetEntity.auditInfo())
+          .build();
+
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to update fileset " + ident, ioe);
+    } catch (NoSuchEntityException nsee) {
+      throw new NoSuchFilesetException("Fileset " + ident + " does not exist", nsee);
+    } catch (AlreadyExistsException aee) {
+      // This is happened when renaming a fileset to an existing fileset name.
+      throw new RuntimeException("Fileset " + ident + " already exists", aee);
```

Comment:
we'd better not use `ident` here. It's quite misleading here. we may use 
`throw new RuntimeException("Fileset to rename already exists", are)....`